light-client/src/operations/voting_power.rs

```diff
@@ -24,39 +83,111 @@ impl VotingPowerCalculator for ProdVotingPowerCalculator {
         validators.total_power()
     }
 
-    fn voting_power_in(
+    fn voting_power_of(
         &self,
         signed_header: &SignedHeader,
-        validators: &ValidatorSet,
-    ) -> Result<u64, BoxError> {
-        // NOTE: We don't know the validators that committed this block,
-        //       so we have to check for each vote if its validator is already known.
-        let mut signed_power = 0_u64;
-
-        for vote in &signed_header.signed_votes() {
-            // Only count if this vote is from a known validator.
-            // TODO: we still need to check that we didn't see a vote from this validator twice ...
-            let val_id = vote.validator_id();
-            let val = match validators.validator(val_id) {
-                Some(v) => v,
-                None => continue,
+        validator_set: &ValidatorSet,
+        trust_threshold: TrustThreshold,
+    ) -> Result<VotingPower, VerificationError> {
+        let signatures = &signed_header.commit.signatures;
+
+        let mut tallied_voting_power = 0_u64;
+        let mut seen_validators = HashSet::new();
+
+        for (idx, signature) in signatures.into_iter().enumerate() {
+            let vote = vote_from_non_absent_signature(signature, idx as u64, &signed_header.commit);
```

Comment:
You mean counting absent signatures? I don't think that's possible since a `CommitSig::BlockIDFlagAbsent` does not contain any of the data required to craft a `Vote` (ie. validator_address, timestamp and signature).

Reply:
True!